CustomLinkedList/src/main/java/org/example/CustomLinkedList.java

```diff
@@ -0,0 +1,173 @@
+package org.example;
+
+public class CustomLinkedList<T> {
+    private CustomLinkedListElement<T> currentElement;
```

Comment:
- is problematic as it serves as both a pointer for operations and as the head/tail reference
- Missing a proper head/tail reference design pattern common in linked lists
- No iterator implementation for traversal